### PR TITLE
Add reputation token trade scheme

### PR DESCRIPTION
**Pull request by ben-kaufman**

### CODE REVIEW

File: contracts/schemes/ReputationTokenTrade.sol [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+            emit TokenTradeProposalExecuted(
+                address(avatar),
+                _proposalId,
+                proposal.beneficiary,
+                proposal.sendToken,
+                proposal.sendTokenAmount,
+                proposal.reputationAmount
+            );
+            delete proposals[_proposalId];
```
Comment (orenyodfat) [inline contracts/schemes/ReputationTokenTrade.sol:106]: can delete proposal be moved out of the if and make it one time

File: contracts/schemes/ReputationTokenTrade.sol [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    * @return bool success
+    */
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    override
+    returns(bool) {
+        Proposal memory proposal = proposals[_proposalId];
```
Comment (orenyodfat) [inline contracts/schemes/ReputationTokenTrade.sol:77]: unused var ?

Comment (orenyodfat) [reply contracts/schemes/ReputationTokenTrade.sol:77]: also at https://github.com/daostack/arc/blob/master-2/contracts/schemes/TokenTrade.sol#L80

File: contracts/schemes/ReputationTokenTrade.sol [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        }
+        proposals[_proposalId].decided = true;
+
+        emit ProposalExecuted(address(avatar), _proposalId, _decision);
+        return true;
+    }
+
+
+    function execute(bytes32 _proposalId) public {
+        Proposal storage proposal = proposals[_proposalId];
```
Comment (orenyodfat) [inline contracts/schemes/ReputationTokenTrade.sol:88]: why this is `storage` ?

File: contracts/schemes/ReputationTokenTrade.sol [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                _proposalId,
+                proposal.beneficiary,
+                proposal.sendToken,
+                proposal.sendTokenAmount,
+                proposal.reputationAmount
+            );
+        } else {
+            proposal.sendToken.safeTransfer(address(proposal.beneficiary), proposal.sendTokenAmount);
+        }
+        delete proposals[_proposalId];
```
Comment (orenyodfat) [inline contracts/schemes/ReputationTokenTrade.sol:108]: does` delete proposal` will work ?

Comment (ben-kaufman) [reply contracts/schemes/ReputationTokenTrade.sol:108]: I changed it to `memory` so I don't think so.

**Pull request merged**